google-cloud-pubsub/lib/google/cloud/pubsub/service.rb

```diff
@@ -204,18 +204,27 @@ def list_subscriptions options = {}
         ##
         # Creates a subscription on a given topic for a given subscriber.
         def create_subscription topic, subscription_name, options = {}
-          subscriber.create_subscription \
-            name:                       subscription_path(subscription_name, options),
-            topic:                      topic_path(topic),
-            push_config:                options[:push_config],
-            ack_deadline_seconds:       options[:deadline],
-            retain_acked_messages:      options[:retain_acked],
-            message_retention_duration: Convert.number_to_duration(options[:retention]),
-            labels:                     options[:labels],
-            enable_message_ordering:    options[:message_ordering],
-            filter:                     options[:filter],
-            dead_letter_policy:         dead_letter_policy(options),
-            retry_policy:               options[:retry_policy]
+          updated_option = construct_create_subscription_options topic, subscription_name, options
+          subscriber.create_subscription **updated_option
+        end
+
+        def construct_create_subscription_options topic, subscription_name, options
+          options[:name] = subscription_path subscription_name, options
```

Comment:
We're still modifying the `options` field here. 
How about we `dup` the `options` here and modify the duplicated one? Just something from the top of my head.